### PR TITLE
New tools and cron modifications

### PR DESCRIPTION
**Pull request by stefaweb**
Various changes in tools directory, CRONs and INSTALL doc for 1.2.0.

- created one uniq cron for mailwatch to be put in /etc/cron.daily
- created a cron for Postfix MTA to be put in /etc/cron.hourly
- moved all .php MailWatch system scripts in /usr/local/bin
- move out from /mailscanner dir postfix_relay.php and mailscanner_relay.php (security)
- renamed various files to make them easier to understand
- renamed Sendmail_queue directory to Sendmail-Exim_queue
- deleted obsolete files
- updated various INSTALL doc
- created entry in UPGRADING file for MailWatch 1.2.0
- Renamed all php files with mailwatch_ at start.
- Modified Sendmail_relay with new name for init script.
- New test for functions.php path
- Updated conf.php.example for new path of mailwatch_sendmail_queue.php

(replace #466)

### CODE REVIEW

**Comment by stefaweb** [comment]
Ready to merge.

**Comment by endelwar** [comment]
(copy and paste from #466 
> Do you want "mailwatch" or "mailwatch_relay" as name?

I prefer minus sign `-` over underscore `_`.
Also filenames should be prefixed by `mailwatch-`, followed, when needed, by something significant and possibly short:

- /etc/cron.d/mailwatch
- /etc/cron.d/mailwatch-postfix-relay
- /etc/cron.d/mailwatch-sendmail-relay
- etc...)

**Comment by stefaweb** [comment]
Will do.

File: UPGRADING [hunk 1]
```diff
@@ -8,6 +8,29 @@ procedure.
 
 You are also stongly advised to read the CHANGELOG first!
```
Comment (endelwar) [inline UPGRADING]: typo: strongly

File: UPGRADING [hunk 2]
```diff
@@ -8,6 +8,29 @@ procedure.
...
 
+UPGRADING FROM 1.0.3 TO 1.2.0
+
+- Stop MailScanner and the MTA (Postfix or Exim).
```
Comment (endelwar) [inline UPGRADING]: Postfix, Exim or Sendmail

File: tools/Cron_jobs/INSTALL [hunk 3]
```diff
@@ -2,53 +2,36 @@
...
+For MailScanner v4: Delete the clean.quarantine cron file.
 
-You will need to edit in conf.php the QUARANTINE_DAYS_TO_KEEP definition.
-
-You may need to edit the quarantine_maint.php to reflect the location of the functions.php file.
+For MailScanner v5: Change the variable for Quarantine Retention to "q_days=0" in /etc/MailScanner/defaults.
 
 --------------------------------------------
-Quarantine Report
+MailScanner Rule Editor (optionnal)
```
Comment (endelwar) [inline tools/Cron_jobs/INSTALL]: typo: optional

**Comment by stefaweb** [comment]
And this #478 ?

File: tools/sudo/INSTALL [hunk 4]
```diff
@@ -1,8 +1,12 @@
+----------------------------------
 INSTALLATION INSTRUCTIONS
--------------------------
+----------------------------------
 
-- edit `mailwatch` file to match web server user (www-data, apache or other) and esecutable path
-- copy edited `mailwatch` file in /etc/sudoers.d/
-- set permission to 0440 (`chmod 440 /etc/sudoers.d/mailwatch`)
+- Edit `mailwatch` file to match web server user (www-data, apache or other) 
+  and change esecutable path if needed.
```
Comment (endelwar) [inline tools/sudo/INSTALL]: typo: executable

**Comment by stefaweb** [comment]
Should be ok.

**Comment by stefaweb** [comment]
@endelwar 

I discovered that we have 2 files with the same name.

tools/Sendmail_queue/mailq.php   (renamed in #487 tools/MailWatch_queue/mailq.php)
mailscanner/mailq.php

What is the difference?

I use the one tools/Sendmail_queue/mailq.php and launch it with cron every minute.

We can may be rename it?
tools/MailWatch_queue/mailwatch_mailq.php

**Comment by asuweb** [comment]
The file in tools/Sendmail_queue/mailq.php is exclusively for those who use sendmail, so a better name would be sendmail_mailq.php or sendmail_queue.php

**Comment by asuweb** [comment]
mailscanner/mailq.php is referenced here -

```
functions.php - line 374
// Display the MTA queue
        // Postfix if mta = postfix
        if ($_SESSION['user_type'] === 'A') {
            if ($mta === 'postfix') {
                if (is_readable($incomingdir) && is_readable($outgoingdir)) {
                    $inq = postfixinq();
                    $outq = postfixallq() - $inq;
                    echo '    <tr><td colspan="3" class="heading" align="center">' . __('mailqueue03') . '</td></tr>' . "\n";
                    echo '    <tr><td colspan="2"><a href="postfixmailq.php">' . __('inbound03') . '</a></td><td align="right">' . $inq . '</td>' . "\n";
                    echo '    <tr><td colspan="2"><a href="postfixmailq.php">' . __('outbound03') . '</a></td><td align="right">' . $outq . '</td>' . "\n";
                } else {
                    echo '    <tr><td colspan="3">' . __('verifyperm03') . ' ' . $incomingdir . ' ' . __('and03') . ' ' . $outgoingdir . '</td></tr>' . "\n";
                }
                // else use mailq which is for sendmail and exim
            } elseif (MAILQ) {
                $inq = database::mysqli_result(dbquery('SELECT COUNT(*) FROM inq WHERE ' . $_SESSION['global_filter']), 0);
                $outq = database::mysqli_result(dbquery('SELECT COUNT(*) FROM outq WHERE ' . $_SESSION['global_filter']), 0);
                echo '    <tr><td colspan="3" class="heading" align="center">' . __('mailqueue03') . '</td></tr>' . "\n";
                echo '    <tr><td colspan="2"><a href="mailq.php?queue=inq">' . __('inbound03') . '</a></td><td align="right">' . $inq . '</td>' . "\n";
                echo '    <tr><td colspan="2"><a href="mailq.php?queue=outq">' . __('outbound03') . '</a></td><td align="right">' . $outq . '</td>' . "\n";
            }
```

**Comment by stefaweb** [comment]
ok, its going a little better now. I've to rename some files/dir in this PR.

But if I use Exim, do I need to use tools/Sendmail_queue/mailq.php and place it in a cron?

**Comment by asuweb** [comment]
We need to review the documentation and the comments in conf.php.example so that they match.

It would seem that for exim, yes you do need to run the sendmail/mailq.php from the cron.  Postfix you don't as it can read the queue directly.

The comments in conf.php

```

**Comment by stefaweb** [comment]
> It would seem that for exim, yes you do need to run the sendmail/mailq.php from the cron. Postfix you don't as it can read the queue directly.

I hope so. I'm running tools/Sendmail_queue/mailq.php with Exim for years!

Ok, I work a little to reorganize all these stuff.

**Comment by stefaweb** [comment]
@asuweb: Could you check the new file organisation in tool/ for Postfix of this PR?

capture d ecran 2017-02-06 a 16 42 10

**Comment by asuweb** [comment]
Seems I rewrote the cron for this a while back on my live server - now I remember why.

With postfix_relay, there are 2 options, run it once per hour with --refresh or tail the log continuously.  

I prefer tailing the log, so I run from cron every hour the following:

```
#!/bin/bash
if ps -C php -o args h | grep postfix_relay.php
then true ## postfix_relay.php running
else /usr/bin/php -q /var/www/html/mailscanner/postfix_relay.php > /dev/null &
fi
if ps -C php -o args h | grep mailscanner_relay.php
then true ## mailscanner_relay.php running
else /usr/bin/php -q /var/www/html/mailscanner/mailscanner_relay.php > /dev/null &
fi

```

But yes, your proposal works, but it would be good to at least mention the purpose of `--refresh` in the install doc so admins can choose what they prefer

**Comment by stefaweb** [comment]
I'll follow that Postfix lover will say. ;)

If you have proposal for tools/Postfix_relay/INSTALL. Just post it here. I will add it.

**Comment by Skywalker-11** [comment]
the mailwatch_postfix_relay wont work this way because the __DIR__ doesn't point to the mailscanner directory (functions.php folder). That path has to be set in the file and the INSTALL should contain a notice to change it.

**Comment by stefaweb** [comment]
Can we use this kind of test?

```
// Edit if you changed webapp directory from default
$pathToFunctions = '/var/www/html/mailscanner/functions.php';

if (!is_file($pathToFunctions)) {
    die('Cannot find functions.php file in "' . $pathToFunctions . '": edit ' . __FILE__ . ' and set the right path on line ' . (__LINE__ - 3) . "\n");
}

require_once $pathToFunctions;
```

**Comment by Skywalker-11** [comment]
you should use `if (!@is_file($pathToFunctions)) {` to prevent the PHP Warning and may prefix the die message with "ERROR"

**Comment by stefaweb** [comment]
Like this?

```
// Edit if you changed webapp directory from default
$pathToFunctions = '/var/www/html/mailscanner/functions.php';

if (!@is_file($pathToFunctions)) {
    die('Error: Cannot find functions.php file in "' . $pathToFunctions . '": edit ' . __FILE__ . ' and set the right path on line ' . (__LINE__ - 3) . "\n");
}

require $pathToFunctions;

// Edit if you changed webapp directory from default
$pathToFunctions = '/var/www/html/mailscanner/mtalogprocessor.inc.php';

if (!@is_file($pathToFunctions)) {
    die('Error: Cannot find mtalogprocessor.inc.php file in "' . $pathToFunctions . '": edit ' . __FILE__ . ' and set the right path on line ' . (__LINE__ - 3) . "\n");
}

require_once $pathToFunctions;
```

For functions.php: require_once or require?

**Comment by Skywalker-11** [comment]
You should mention in tools/Postfix_relay/INSTALL that they have to change the path

**Pull request merged**